src/server/game/Maps/Map.cpp

```diff
@@ -913,6 +934,8 @@ struct ResetNotifier
 
 void Map::RemovePlayerFromMap(Player* player, bool remove)
 {
+    // Before leaving map, update zone/area for stats
+    player->UpdateZone(MAP_INVALID_ZONE, 0);
```

Comment:
Can we not call `UpdatePlayerZoneStats(m_zoneUpdateId, newZone);` directly?

Reply:
wont have time to investigate, could be investigated later, this is a cherry-pick, so we should be safe